lib/logger/lib/logger/backends/console.ex

```diff
@@ -236,7 +236,13 @@ defmodule Logger.Backends.Console do
   defp retry_log(error, %{device: device, ref: ref, output: dirty} = state) do
     Process.demonitor(ref, [:flush])
 
-    case :unicode.characters_to_binary(dirty) do
+    try do
+      :unicode.characters_to_binary(dirty)
+    rescue
+      ArgumentError ->
+        clean = ["failure while trying to output malformed data: ", inspect(dirty), ?\n]
+        %{state | ref: async_io(device, clean), output: clean}
```

Comment:
I am definitely not sure about this, and this is why I didn't add tests yet (would prefer to get feedback on the approach first). One thing that bugs me is that if we get here, the log will be weird in the console:

```
iex(1)> Logger.configure(truncate: :infinity)
:ok
iex(2)> Logger.bare_log(:info, [%{}])
:ok
iex(3)> failure while trying to output malformed data: [[[] | "\e[22m"], ["\n", [[48, "0"], 58, "40", 58, "57", 46, "292"], " ", "", "[", "info", "] ", " ", [%{}], "\n"] | "\e[0m"]
nil
```

Thoughts?

Reply:
Something else we could do is to perform an asynchronous call to the logger so it is processed as a regular log message.

Reply:
@josevalim but then it would be broadcast to all backends, while there is a problem only with the console backend. Is it fine to log this to all backends?